Source/WebKit/NetworkProcess/Downloads/cocoa/DownloadCocoa.mm

```diff
@@ -78,18 +78,41 @@
 void Download::platformDestroyDownload()
 {
     if (m_progress)
-#if HAVE(NSPROGRESS_PUBLISHING_SPI)
+#if HAVE(NSPROGRESS_PUBLISHING_SPI) && !HAVE(MODERN_DOWNLOADPROGRESS)
         [m_progress _unpublish];
 #else
         [m_progress unpublish];
 #endif
+#if HAVE(MODERN_DOWNLOADPROGRESS)
+    m_bookmarkURL = nil;
+#endif
 }
 
 #if HAVE(MODERN_DOWNLOADPROGRESS)
 void Download::publishProgress(const URL& url, std::span<const uint8_t> bookmarkData)
 {
     RetainPtr bookmark = toNSData(bookmarkData);
-    m_progress = adoptNS([[WKDownloadProgress alloc] initWithDownloadTask:m_downloadTask.get() download:*this URL:(NSURL *)url bookmarkData:bookmark.get()]);
+    m_bookmarkData = bookmark;
+
+    BOOL bookmarkIsStale = NO;
+    NSError* error = nil;
+    m_bookmarkURL = adoptNS([NSURL URLByResolvingBookmarkData:m_bookmarkData.get() options:NSURLBookmarkResolutionWithoutUI relativeToURL:nil bookmarkDataIsStale:&bookmarkIsStale error:&error]);
+    if (!m_bookmarkURL)
+        NSLog(@"Unable to create bookmark URL, error = %@", error);
+
+    bool enableModernDownloadProgress = CFPreferencesGetAppBooleanValue(CFSTR("EnableModernDownloadProgress"), CFSTR("com.apple.WebKit"), NULL);
```

Comment:
boolean variables need a prefix in WebKit. Here maybe `shouldEnableModernDownloadProgress`

Reply:
Ah, good point, will fix!

Thanks for reviewing!